python/mleap/pyspark/feature.py

```diff
@@ -0,0 +1,38 @@
+from pyspark.ml.util import JavaMLReadable, JavaMLWritable, _jvm
+from pyspark.ml.wrapper import JavaTransformer
+from pyspark.ml.param.shared import HasInputCol, HasOutputCol
+
+
+class StringMap(JavaTransformer, HasInputCol, HasOutputCol, JavaMLReadable, JavaMLWritable):
+    def __init__(self, labels, inputCol=None, outputCol=None, handleInvalid='error', defaultValue=0.0):
+        """
+        __init__(self, labels, inputCol=None, outputCol=None, handleInvalid='error', defaultValue=0.0)
+        labels is a dict {string: double}
+        handleInvalid: how to handle missing labels: 'error' (throw an error), or 'keep' (map to the default value)
+        """
+        assert handleInvalid in ['error', 'keep'], 'Invalid value for handleInvalid: {}'.format(handleInvalid)
+        super(StringMap, self).__init__()
+        labels_scala_map = _jvm() \
+            .scala \
+            .collection \
+            .JavaConverters \
+            .mapAsScalaMapConverter(labels) \
+            .asScala() \
+            .toMap(_jvm().scala.Predef.conforms())
+        handle_invalid_jvm = _jvm().ml.combust.mleap.core.feature.HandleInvalid.__getattr__(
+            handleInvalid.capitalize() + '$').__getattr__('MODULE$')
+        string_map_model = self._new_java_obj("ml.combust.mleap.core.feature.StringMapModel",
+                                              labels_scala_map, handle_invalid_jvm, defaultValue)
+        self._java_obj = self._new_java_obj("org.apache.spark.ml.mleap.feature.StringMap", self.uid, string_map_model)
+        self.setInputCol(inputCol)
+        self.setOutputCol(outputCol)
+
+    @classmethod
+    def from_dataframe(cls, df, key_col, value_col):
```

Comment:
when/how is this method going to be used/needed?

Reply:
It has been used to convert a regular spark DataFrame into a a string transformer. I think it's a common use case. For example:
- Prepare some mappings with spark. eg. most simple case: `df = spark.table('some.table')`
- Converting the result into a python dict is not _that_ trivial, so having a helper for it makes sense.
- It allows keeping the code a bit cleaner, ie. this one-liner:
	```python
	string_map = StringMap.from_dataframe(df, 'key_col', 'value_col')`
	```
  instead of this:
	```python
	rows = df.select(['key_col', 'value_col']).collect()
	mappings_dict = {row[0]: float(row[1]) for row in rows}
	string_map = StringMap(mappings_dict, 'key_col', 'value_col')
	```

However, maybe the API could be different? Another option would be to check the type of `labels` in `__init__`, and if it's a `DataFrame`, collect it into a dict. How would you like that?

Reply:
I think it's okay as is, I just wanted to understand a bit better its use :)